Add 'Considerations When Using Kubeadm' section to k8s quickstart (#196)
spiffe/spiffe.io: content/docs/latest/spire/installing/getting-started-k8s.md

{{< info >}}
If you are using Minikube to run this tutorial you should specify some special flags as described [here](#considerations-when-using-minikube).
{{< /info >}}
{{< info >}}

> If you are using Kubeadm to run this tutorial, you should create the resources specified [here](#considerations-when-using-kubeadm).

Author: I'm not sure if this is 100% correct, because we are creating a storage class and a storage provisioner to bind a claim to an automatically created persistent volume. Maybe we can write something like:
```suggestion
If you are using Kubeadm to run this tutorial, you should manually create the resources needed to provide a persistent volume, as described [here](#considerations-when-using-kubeadm).
```
What do you think?

> If you are using Kubeadm to run this tutorial, you should create the resources specified [here](#considerations-when-using-kubeadm).

Reviewer: 👍  Yes, that sounds good. I'm glad to have it as accurate as possible.

For Kubernetes versions prior to 1.17.0 the `apiserver.authorization-mode` can be specified as `apiserver.authorization-mode=RBAC`.
{{< /info >}}

# Considerations When Using Kubeadm

> If you are using Kubeadm to run this tutorial, a default storage class and an associated provisioner must be manually created. To achieve this, change to the directory named **kubeadm**, and run the two commands shown below.

Author: I agree in this, it makes the workflow more clear. Just a little modification:
```suggestion
If you are using Kubeadm to run this tutorial, a default storage class and an associated provisioner must be manually created. To achieve this, proceed with the steps above, and after running the command in [Create Server Bundle Configmap, Role & ClusterRoleBinding](#create-server-bundle-configmap-role--clusterrolebinding) section, change to the directory named **kubeadm**, and run the two commands shown below.
```

> If you are using Kubeadm to run this tutorial, a default storage class and an associated provisioner must be manually created. To achieve this, change to the directory named **kubeadm**, and run the two commands shown below.

Reviewer: Sure, that's fine. One last tweak to fix a problem introduced in my version, let's omit "section". ("Section" would be needed if there was a "the" at the beginning of the clause.) So let's make the last sentence:

_To achieve this, proceed with the steps above, and after running the command in [Create Server Bundle Configmap, Role & ClusterRoleBinding](#create-server-bundle-configmap-role--clusterrolebinding), change to the directory named **kubeadm**, and run the two commands shown below._

``` 
$ kubectl apply -f default-storage-class.yaml
$ kubectl apply -f storage-provisioner.yaml 
```

> {{< info >}}

Author: Good idea! I'm thinking that maybe this note can be placed outside of the info brackets, and only keep the clarification about the moment in which the resources need to be applied inside it.
 ```suggestion
Continue with the steps in [Create Server Configmap](#create-server-configmap).
{{< info >}}
If these resources are not applied before the **spire-server** statefulset creation, the pod will stay in `Pending` status.
{{< /info >}}
```
I'm open to suggestions

> {{< info >}}

Reviewer: 👍  Yes, good idea.

> These commands must be executed before spire-server statefulset creation, otherwise the pod will stay in `Pending` status

Author: Answered below (sentence changed a little bit)
